engine/dlib/src/dlib/buffer.cpp

```diff
@@ -616,4 +644,117 @@ namespace dmBuffer
         buffer->m_ContentVersion++;
         return RESULT_OK;
     }
+
+    // Returns a pointer to the metadata item or 0 if not found.  Assumes buffer is valid.
+    Buffer::MetaData* FindMetaDataItem(Buffer* buffer, dmhash_t name_hash) {
+        dmArray<Buffer::MetaData*>& arr = buffer->m_MetaDataArray;
+        for (uint32_t  i=0; i<arr.Size(); i++) {
+            if (arr[i]->m_Name == name_hash) {
+                return arr[i];
+            }
+        }
+        return 0; // nothing found
+    }
+
+
+    Result GetMetaData(HBuffer hbuffer, dmhash_t name_hash, void** data, uint32_t* count, ValueType* type) {
+        Buffer* buffer = GetBuffer(g_BufferContext, hbuffer);
+        if (!buffer) {
+            return RESULT_BUFFER_INVALID;
+        }
+
+        Buffer::MetaData* item = FindMetaDataItem(buffer, name_hash);
+        if (!item) {
+            return RESULT_METADATA_NOT_EXIST;
+        }
+
+        *count = item->m_ValueCount;
+        *type = (ValueType)item->m_ValueType; // assumes value tpe is sanitized when setting the metadata
```

Comment:
"tpe" -> "type"

Not sure that the comment refers to?

Reply:
`item->m_ValueType`  is a `uint8_t`, and we're typecasting to ValueType. 

We're _assuming_ that is has a proper value that indeed matches to ValueType. And we're doing that because we know that we set it right.  I've updated the comment to describe this better.